### PR TITLE
Vote locks for all reasons except RESERVE

### PR DESCRIPTION
**Pull request by muharem**
Votes and delegations should lock for all reasons except RESERVE.

The changes do not change anything for clients who uses the latest version of balances pallet (Currency and fungibles from it). Since those implementations ignore the reason.

### CODE REVIEW

File: frame/conviction-voting/src/lib.rs [hunk 1]
```diff
@@ -669,7 +674,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				CONVICTION_VOTING_ID,
 				who,
 				lock_needed,
-				WithdrawReasons::TRANSFER,
+				WithdrawReasons::TRANSFER | WithdrawReasons::TIP,
```
Comment (muharem) [inline frame/conviction-voting/src/lib.rs:677]: not sure if we can make this `WithdrawReasons::all()`

**Review by joepetrowski — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2720183

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2720182

**Review by gilescope — APPROVED** [review]

**Comment by muharem** [comment]
bot merge

**Pull request merged**